src/dist/dist.rs

```diff
@@ -459,6 +459,46 @@ impl<'a> Manifest<'a> {
     }
 }
 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub enum Profile {
+    Minimal,
+    Default,
+    Complete,
+}
+
+impl Profile {
+    pub fn from_str(name: &str) -> Result<Self> {
+        match name {
+            "minimal" | "m" => Ok(Profile::Minimal),
+            "default" | "d" | "" => Ok(Profile::Default),
+            "complete" | "c" => Ok(Profile::Complete),
+            _ => Err(ErrorKind::InvalidProfile(name.to_owned()).into()),
+        }
+    }
+
+    pub fn names() -> &'static [&'static str] {
+        &["minimal", "default", "complete"]
+    }
+
+    pub fn default_name() -> &'static str {
+        "default"
+    }
+
+    // Non-required components that were required before profiles exist.
+    pub fn legacy_profile() -> Vec<String> {
+        ["rust-std", "rustc", "cargo", "rust-docs"]
```

Comment:
I think we should add rust-mingw here (it should match what we're doing now by default). We might need to handle it not being present on some platforms, but I don't think that will cause big problems.

Reply:
Okay, this means that we also need to decide what we're actually going to need to do regarding `rust-mingw` in the real world.  Currently we fail to install any toolchain other than a mingw one because that component is *only* available for the mingw targets.